tests/Fixer/FunctionNotation/NamedArgumentSpaceFixerTest.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\FunctionNotation;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\FunctionNotation\NamedArgumentSpaceFixer
+ *
+ * @requires PHP 8.0
+ */
+final class NamedArgumentSpaceFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @dataProvider provideFixCases
+     */
+    public function testFix(string $expected, ?string $input = null): void
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public static function provideFixCases(): iterable
+    {
+        yield [
+            '<?php foo1(a: 1);',
+            '<?php foo1(a   : 1);',
+        ];
+
+        yield [
+            '<?php foo2(a: 1);',
+            '<?php foo2(a:1);',
+        ];
+
+        yield [
+            '<?php foo3(a: 1);',
+            '<?php foo3(a:    1);',
+        ];
+
+        yield [
+            '<?php foo4(a/* X */    : 1);',
```

Comment:
I am not sure if this is what it should look like, I would rather expect single space around the comment, or at least before it. Multiple spaces between comment and `:` is not what I would like to keep.

Reply:
If people put comments or PHPDocs _between_ the name and the colon than they are on their own. The tests are only added to prove the fixer doesn't break. I'm not going to add logic to support any CS around this case.

Reply:
Fair enough, but then I am not sure if we should add such test case because it will act as a contract and if anybody would improve the fixer to handle comments better we would break that contract.

Reply:
The unit test is not a contract, it is just proves the fixer won't break down and that it will not break the input code.

Reply:
IMHO fixers' test cases are the only possible behavioral contract between us and the users, and when we declare some output for the fixer, we should respect it later, not juggle it freely. If we only want to ensure that input code does is not broken by the fixer, we should use at least different data provider or integration test (marked as not covered by BC promise) 🙂.

I don't know if there was such approach before, but for every issue reported by users the first thing I check is if there are test cases for such scenario. If we introduce test case that ensure we format such comments in one way, we probably shouldn't change it later.

Would be great to hear others' opinion 🙂.

Reply:
The fixer now fixes:

```
<?php foo4(a     /* X */    :    1);
```

into:

```
<?php foo4(a/* X */    : 1);
```

and your BC concerns are that someone in the future may fix it to:

```
<?php foo4(a /* X */: 1);
```

and the users are disappointed because they liked 

```
<?php foo4(a/* X */    : 1);
```

better?

Reply:
It's not exactly about what they like better, but about the need to react to changes enforced by the tool in the future 🙂. You know, users introduce this fixer in their ruleset, apply it on the whole codebase, then at some point they update Fixer that provides new way of handling comments and now their CI fails because new standard has to be applied. In general it's not a big deal, but I would rather not add this test case so there's no explicit contract for this scenario, than adding it now and deal with light BC break or the need for introducing config option for that later. 

Anyway, ideal scenario would be producing `<?php foo4(a /* X */: 1);` when fixer is applied, since it's sane default, but if you don't want to provide it in this PR, then I would drop test cases with comments completely, which would explicitly state that such usage is not covered with our BC policy.

But that's only my opinion, I really would like to know what others' think 🙂.

Reply:
We have a practice to add doc/comment in any random places that we wouldnd't normally imagine to use them - especially to ensure fixer will not explode on it (or produce invalid PHP syntax) - because _we had such cases_.

:+1: for having this tests.

about "contract or not" - we often say that "for rule X, this is how we fix currently - covering variant A and B, but not yet C - and with new version of rule, it may start fixing variant C as well". If the change to handle case C was significant enough, we were introducing it behind rule option and disabled by default. That way, you keep the contract but you also ensure you don't have a rule that explodes on unexpected syntax.

We could also explicitly write comment next to the test case "crazy syntax, do not touch for now" - and if one would ever (myself :-1:) be eager to start fixing such syntax, they have a start point.